modules/storage/storage.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ExtraVolumePropagation defines a const that represent the propagation policy
+type ExtraVolumePropagation string
+// ExtraVolType represents a "label" that can be optionally added to the VolMounts
+// instance
+type ExtraVolType string
+// ServiceType identifies the Service, Group or instance (e.g. the backend) that
+// receives an Extra Volume that can potentially be mounted
+type ServiceType string
+
+const (
+	// PropagationEverywhere is used to define a propagation policy that allows
+	// to get the volumes mounted to all the OpenStack services
+	PropagationEverywhere ExtraVolumePropagation = "All"
+	// DBSync represents a common ServiceType defined by the OpenStack operators
+	// that keeps track of the DBSync pod
+	DBSync                ServiceType            = "DBSync"
+	// Compute represents a common ServiceType that can be translated into an
+	// external-data-plane related propagation policy
+	Compute               ServiceType            = "Compute"
```

Comment:
Is this specific for Computes, since there can be other external data plane components. would they their own  ServiceType or is an "External" ServiceType generic enough?

Reply:
For `Compute` I mean the `external-data-plane` which is supposed to see files propagated via `ansible-operator`, which is the only external component I'm aware so far.
We can change it to `External` if you think that `Compute` can result confusing and it doesn't fit well for the use case, but in theory I think the only node type outside of OCP will be Compute (managed by the jobs triggered by the `ansible-operator`).
Let me know your thoughts, this can be an easy change while we're in the middle of reviewing the lib-common/storage content.

Reply:
At this stage Compute works, but I agree we should align and follow up here according to the evolution of this work: `External` might not be a good fit if we still need to propagate volumes in a more granular way, but of course, given that `Nova` can adopt this method, we should address the requirements of that use case.